tests/mock_callables.py

```diff
@@ -0,0 +1,97 @@
+"""
+mocks.py - Mock callables. Objects that can be called and later inquired as to how they were called.
+
+Copyright (c) 2018 The Fuel Rats Mischief,
+All rights reserved.
+
+Licensed under the BSD 3-Clause License.
+
+See LICENSE.
+"""
+from typing import List, NamedTuple, Dict, Any, Tuple
+
+_Call = NamedTuple("_Call", args=tuple, kwargs=Dict[str, Any])
+
+
+class InstanceOf(object):
+    """
+    Simple wrapper around a class. To be used in conjunction with :class:`CallableMock` instances.
+    Instances of :class:`InstanceOf` compare equal with instances of their wrapped type.
+
+    Examples:
+        >>> InstanceOf(str) == "hello"
+        True
+        >>> InstanceOf(dict) == 7
+        False
+    """
+    __slots__ = ("klass",)
+
+    def __init__(self, klass: type):
+        self.klass = klass
+
+    def __eq__(self, other):
+        if isinstance(other, self.klass):
+            return True
+        elif isinstance(other, InstanceOf):
+            return self.klass is other.klass
+        else:
+            return NotImplemented
+
+
+class CallableMock(object):
```

Comment:
if we need something akin magicmock from `unittest` we can make use of the [pytest-mock](https://github.com/pytest-dev/pytest-mock) library. this may work better than implementing our own solution.

Reply:
Much of pytest-mock is simply a wrapper around unittest, in fact, `mocker.stub()` seems to just return a `MagicMock` object, which you didn't want me using.

I also am not a huge fan of `MagicMock`, so I've implemented this much simpler and in my opinion easier to use module. It also makes it so much easier to have mock coroutine functions.